src/algorithms/path.c

```diff
@@ -0,0 +1,58 @@
+/*
+* Copyright 2018-2019 Redis Labs Ltd. and Contributors
+*
+* This file is available under the Apache License, Version 2.0,
+* modified with the Commons Clause restriction.
+*/
+
+#include "./path.h"
+#include "../util/arr.h"
+
+Path Path_new(size_t len) {
+    return array_new(Edge*, len);
+}
+
+Path Path_append(Path p, Edge *e) {
+    return array_append(p, e);
+}
+
+Edge *Path_pop(Path p) {
+    size_t pathLen = array_len(p);
+    if(pathLen == 0) return NULL;
+    return array_pop(p);
+}
+
+size_t Path_len(const Path p) {
+    return array_len(p);
+}
+
+Path Path_clone(const Path p) {
```

Comment:
It seems to me like we could just perform a memcpy on the array object to accomplish this in one instruction (arr.h provides an `array_sizeof` macro already). That's not particularly important, though, we can just keep it in mind for the future.